sale_partner_incoterm/models/sale_order.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2015 Opener B.V. (<https://opener.am>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        res = super().onchange_partner_id()
+        if not self.partner_id:
+            self.incoterm = False
+            return res
+        self.incoterm = self.partner_id.sale_incoterm_id
```

Comment:
We don't need this code
```
        if not self.partner_id:
            self.incoterm = False
            return res
```
As when partner_id is not available then self.partner_id.sale_incoterm_id will result as False by default with the New ORM concept of recordset